projects/pastanaga/src/lib/textfield/textarea.component.ts

```diff
@@ -16,6 +21,41 @@ import { TextfieldCommon } from './textfield.common';
         multi: true,
     }],
 })
-export class TextareaComponent extends TextfieldCommon {
+export class TextareaComponent extends TextfieldCommon implements OnInit {
     baseId = 'textarea';
+    keyUpDebouncer: Subject<string> = new Subject();
+    @ViewChild('element', { static: true }) textarea?: ElementRef;
+
+    constructor(private renderer: Renderer2) {
+        super();
+    }
+
+    ngOnInit() {
+        this.keyUpDebouncer.pipe(
+            throttleTime(300),
+        ).subscribe((text) => this.updateTextareaSize(text));
+    }
+
+    onKeyUp($event) {
+        if (!!$event.target && !!$event.target.value) {
+            this.keyUpDebouncer.next($event.target.value);
+        }
+        super.onKeyUp($event);
+    }
+
+    private updateTextareaSize(text: string) {
+        if (!!this.textarea) {
+            const textarea = this.textarea.nativeElement;
+            const linesCount = text.split('\n').reduce((count, chunk, index) => {
+                let newLinesCount = index > 0 ? 1 : 0;
+                if (chunk.length >= TEXTAREA_LINE_LENGTH) {
+                    newLinesCount ++;
```

Comment:
What happens if the line triples or quadruples the default line-length? We should do `newLinesCount += 3` or `newLinesCount += 4` right? I'm not sure about this

Reply:
I don't think that could happen as the function is triggered quite often 🤔

Reply:
As I understand, we are calculating the high of the textarea using the number of `\n` characters  and if that line has more than `TEXTAREA_LINE_LENGTH`(72) characters we add another space because that line will overflow to a second line, but what happens if that line is so long that it will overflow to 3 or 4 lines? We do not control that, I think